tests/return_path_analyser.cpp

```diff
@@ -0,0 +1,100 @@
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "ast/passes/field_analyser.h"
+#include "ast/passes/return_path_analyser.h"
+#include "ast/passes/semantic_analyser.h"
+#include "clang_parser.h"
+#include "driver.h"
+#include "mocks.h"
+
+namespace bpftrace {
+namespace test {
+namespace return_path_analyser {
+
+using ::testing::_;
+
+void test(BPFtrace &bpftrace, const std::string &input, int expected_result = 0)
+{
+  Driver driver(bpftrace);
+  std::stringstream out;
+  std::stringstream msg;
+  msg << "\nInput:\n" << input << "\n\nOutput:\n";
+
+  ASSERT_EQ(driver.parse_str(input), 0);
+
+  ast::FieldAnalyser fields(driver.root.get(), bpftrace, out);
+  ASSERT_EQ(fields.analyse(), 0) << msg.str() << out.str();
+
+  ClangParser clang;
+  ASSERT_TRUE(clang.parse(driver.root.get(), bpftrace));
+
+  ASSERT_EQ(driver.parse_str(input), 0);
+  out.str("");
+  ast::SemanticAnalyser semantics(driver.root.get(), bpftrace, out, false);
+  ASSERT_EQ(semantics.analyse(), 0) << msg.str() << out.str();
+
+  ast::ReturnPathAnalyser return_path(driver.root.get(), out);
+  EXPECT_EQ(return_path.analyse(), expected_result) << msg.str() << out.str();
+}
+
+void test(const std::string &input, int expected_result = 0)
+{
+  auto bpftrace = get_mock_bpftrace();
+  bpftrace->feature_ = std::make_unique<MockBPFfeature>(true);
+  test(*bpftrace, input, expected_result);
+}
+
+TEST(return_path_analyser, simple_return)
+{
+  test("fn test(): int64 { $x = 0; return 0; }", 0);
+}
+
+TEST(return_path_analyser, simple_no_return)
+{
+  test("fn test(): int64 { $x = 0; }", 1);
+}
+
+TEST(return_path_analyser, if_else)
+{
+  test("fn test($x: int64): int64 { if ($x > 0) { return 0; } else { return 1; "
+       "} }",
```

Comment:
Can we do something with the ugly linebreaks or is it clang-format enforcing this?
```suggestion
  test("fn test($x: int64): int64 {"
       "  if ($x > 0) { return 0; } else { return 1; }"
       "}",
```
Same below.

Reply:
I don't think clang-format is enforcing this, I just let it auto-format the file and it did it like that.